File: backend/src/main/java/com/rashid/backend/service/interfaces/TeamService.java
package com.rashid.backend.service.interfaces;

import com.rashid.backend.dto.team.TeamDTO;
import com.rashid.backend.dto.team.TeamInviteDTO;
import com.rashid.backend.dto.team.TeamInviteRequestDTO;
import com.rashid.backend.dto.team.TeamMemberDTO;
import com.rashid.backend.dto.team.TeamRoleUpdateDTO;

import java.util.List;

public interface TeamService {
    List<TeamDTO> getUserTeams(String username);
    TeamDTO createTeam(TeamDTO request, String username);
    TeamDTO updateTeam(Long teamId, TeamDTO request, String username);
    List<TeamMemberDTO> getTeamMembers(Long teamId, String username);
    TeamInviteDTO createInvite(Long teamId, TeamInviteRequestDTO request, String username);
    List<TeamInviteDTO> getTeamInvites(Long teamId, String username);
    List<TeamInviteDTO> getMyPendingInvites(String username);
    TeamInviteDTO acceptInvite(Long inviteId, String username);
    TeamInviteDTO declineInvite(Long inviteId, String username);
    TeamMemberDTO updateMemberRole(Long teamId, Long userId, TeamRoleUpdateDTO request, String username);
    void removeMember(Long teamId, Long userId, String username);
}

File: frontend/src/services/teamService.js
import axios from 'axios'

export const getTeams = async () => (await axios.get('/api/teams')).data
export const createTeam = async (payload) => (await axios.post('/api/teams', payload)).data
export const updateTeam = async (teamId, payload) => (await axios.put(`/api/teams/${teamId}`, payload)).data
export const getTeamMembers = async (teamId) => (await axios.get(`/api/teams/${teamId}/members`)).data
export const getTeamInvites = async (teamId) => (await axios.get(`/api/teams/${teamId}/invites`)).data
export const getMyInvites = async () => (await axios.get('/api/teams/invites/me')).data
export const createInvite = async (teamId, username) => (await axios.post(`/api/teams/${teamId}/invites`, { username })).data
export const acceptInvite = async (inviteId) => (await axios.post(`/api/teams/invites/${inviteId}/accept`)).data
export const declineInvite = async (inviteId) => (await axios.post(`/api/teams/invites/${inviteId}/decline`)).data
export const updateMemberRole = async (teamId, userId, role) => (await axios.put(`/api/teams/${teamId}/members/${userId}/role`, { role })).data
export const removeMember = async (teamId, userId) => axios.delete(`/api/teams/${teamId}/members/${userId}`)
